console-subscriber/tests/support/task.rs

```diff
@@ -0,0 +1,228 @@
+            if expected_self_wakes != actual_task.self_wakes {
+                return Err(TaskValidationFailure {
+                    expected: self.clone(),
+                    actual: Some(actual_task.clone()),
+                    failure: format!(
+                        "{self}: expected `self_wakes` to be {expected_self_wakes}, but actual was {actual_self_wakes}",
+                        actual_self_wakes = actual_task.self_wakes),
+                });
+            }
+        }
+
+        if no_expectations {
+            return Err(TaskValidationFailure {
+                expected: self.clone(),
+                actual: Some(actual_task.clone()),
+                failure: format!(
+                    "{self}: no expectations set, if you want to just expect that a matching task is present, use `expect_present()`")
+            });
+        }
+
+        Ok(())
+    }
+
+    /// Matches tasks by name.
+    ///
+    /// To match this expected task, an actual task must have the name `name`.
+    #[allow(dead_code)]
+    pub(crate) fn match_name(mut self, name: String) -> Self {
+        self.match_name = Some(name);
+        self
+    }
+
+    /// Matches tasks by the default task name.
+    ///
+    /// To match this expected task, an actual task must have the default name
+    /// assigned to the task which runs the future provided to [`assert_task`]
+    /// or [`assert_tasks`].
+    ///
+    /// [`assert_task`]: fn@support::assert_task
+    /// [`assert_tasks`]: fn@support::assert_tasks
+    #[allow(dead_code)]
+    pub(crate) fn match_default_name(mut self) -> Self {
+        self.match_name = Some(MAIN_TASK_NAME.into());
+        self
+    }
+
+    /// Expects that a task is present.
+    ///
+    /// To validate, an actual task matching this expected task must be found.
+    #[allow(dead_code)]
+    pub(crate) fn expect_present(mut self) -> Self {
+        self.expect_present = Some(true);
+        self
+    }
+
+    /// Expects that a task has a specific value for `wakes`.
+    ///
+    /// To validate, the actual task matching this expected task must have
+    /// a count of wakes equal to `wakes`.
+    #[allow(dead_code)]
+    pub(crate) fn expect_wakes(mut self, wakes: u64) -> Self {
+        self.expect_wakes = Some(wakes);
+        self
+    }
+
+    /// Expects that a task has a specific value for `self_wakes`.
+    ///
+    /// To validate, the actual task matching this expected task must have
+    /// a count of self wakes equal to `self_wakes`.
+    #[allow(dead_code)]
+    pub(crate) fn expect_self_wakes(mut self, self_wakes: u64) -> Self {
+        self.expect_self_wakes = Some(self_wakes);
+        self
+    }
+}
+
+impl fmt::Display for ExpectedTask {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let fields = match &self.match_name {
+            Some(name) => format!("name={name}"),
+            None => "(no fields to match on)".into(),
+        };
+        write!(f, "Task<{fields}>")
```

Comment:
nit, TIOLI: using angle brackets here feels a little weird to me, since i would expect them to represent a type parameter? maybe:
```suggestion
        write!(f, "Task< {{ {fields} }}")
```